### DISCUSSION TITLE
Sort by occurence at other model from implicit many-to-many relation

### DISCUSSION THREAD

**Discussion by bennettdams** [prisma/prisma#7509] [Q&A] [answered]
This is more an SQL question than a Prisma question, but maybe someone can help.

Imagine a schema with a `Post`, where each one can have multiple `PostTag`s.
`PostTags` are are user-generated and can be freely attached to a `Post`.

```prisma
model Post {
  id             String        @id @default(cuid())
  createdAt      DateTime      @default(now())
  updatedAt      DateTime      @updatedAt
  title          String
  // 
  tags           PostTag[]
}

model PostTag {
  id          String   @id @default(cuid())
  createdAt   DateTime @default(now())
  updatedAt   DateTime @updatedAt
  name        String
  // 
  posts       Post[]
}
```

Now, when a user selects the tags for his post, I want to provide him with some pre-filtered selection (which also gets filtered via input). I wanted to try to do this without Elasticsearch etc. first, maybe it is performant enough.

To make this work in any way whatsoever, I need to sort the tags I'm showing in the pre-selection. I obviously need to filter by the name, but at first I want to sort them by occurrence at other posts.

- Post 1
  - Tag A
  - Tag C
- Post 2
  - Tag A
  - Tag B
  - Tag C
- Post 3
  - Tag A

Occurences:
Tag A: 3
Tag C: 2
Tag B: 1

With this sorting I can priotize what I'm showing in the selection to the user.

**Is there a built-in way to do that with Prisma or do I need to fall back to raw SQL/do it in TypeScript?**

I could also execute this query in a scheduled way and save the result in the database, kinda like a "view". But, out of curiosity, I would still like to know if there's a way to do this with Prisma.

**Comment by ryands17** [comment, marked as answer]
@bennettdams 👋 

Does [this](https://www.prisma.io/docs/concepts/components/prisma-client/filtering-and-sorting/#sort-by-relation-aggregate-value-preview) satisfy your use case?

**Reply by bennettdams** [reply]
Don't know how I missed that, it's perfect. Thank you!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
